### ISSUE TITLE
some question about loss function

### ISSUE THREAD

**Issue by Glutaredoxin** [lehduong/poodle#1]
Hi~
thank you for your code ^^
I found that loss function below is only used in evaluate, and in training use cross-entropy to train base class.
![image](https://user-images.githubusercontent.com/55617926/173177293-7f2df03e-ffc4-42fa-a7f9-3640f06043bf.png)
maybe i misunderstood, but in this way, it doesn't do anything for training.
can you help me get over :)

**Comment by lehduong** [comment]
Thanks for your interest in our work.

Sorry for not making this explicit in the paper but Poodle is a post-processing technique. We plug it into a pre-trained model (e.g., simple/rot/rot+kd as in our paper) when classifying novel classes to improve the performance compared to vanilla [nearest-neighbor classification](https://github.com/lehduong/poodle/blob/master/src/train.py#L475).

**Comment by Glutaredoxin** [comment]
thank you for your reply, and sorry to bother you again~
there is my understanding about training process:
1. use base classes(train set) to train feature extractor(e.g. Rsenet12), and then use this feature extractor to evaluate few-shot learning (from novel classes like val set).
2. use KD to train student model in base classes which is similar to above and use student model to evaluate few-shot learning in novel classes which is similar to above.
3. when training is over, use poodle loss to improve accuracy, but poodle loss will not act on feature extractor~
if it have any problem, please do point out for me. 
thank you again :)

**Comment by lehduong** [comment]
Yes, your understanding is correct.

**Comment by Glutaredoxin** [comment]
ok, i got it !!
thank you very much :)
